dist/tools/compile_commands/compile_commands.py

```diff
@@ -318,11 +318,21 @@ def generate_compile_commands(args):
     if _args.clangd:
         _args.add_built_in_includes = True
         _args.add_libstdcxx_includes = True
-        _args.filter_out = ['-mno-thumb-interwork',
-                            # Only even included for versions of GCC that support it
-                            '-misa-spec=2.2',
-                            '-malign-data=natural',
-                            # Only supported starting with clang 11
-                            '-msmall-data-limit=8',
-                            ]
+        flags = [
+            '-mno-thumb-interwork',
+            # Only even included for versions of GCC that
+            # support it
+            '-misa-spec=2.2',
+            '-malign-data=natural',
+            # Only supported starting with clang 11
+            '-msmall-data-limit=8',
+            # not supported by clang, see
+            # https://gcc.gnu.org/onlinedocs/gcc-10.2.0/gcc/Xtensa-Options.html
+            '-mtext-section-literals',
+            '-fstrict-volatile-bitfields',
+            # it's called -mlong-calls in LLVM, but we don't need it for clangd
+            # as we do not generate code anyway
+            '-mlongcalls',
+        ]
+        _args.filter_out.extend(flags)
```

Comment:
I think this also fixes a bug drive-by, as previously arguments provided by `--filter-out` were overwritten when using `--clangd`, rather than extended.